http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/NonJsonPayLoadIT.java

```diff
@@ -0,0 +1,176 @@
+            "      cities:\r\n" +
+            "        - name: Tokio\r\n" +
+            "          country: Japan\r\n";
+
+    @Test
+    public void testExpectedXmlResponse() {
+        app.given()
+                .get("/city").then()
+                .statusCode(HttpStatus.SC_OK)
+                .contentType(MediaType.APPLICATION_XML)
+                .body("cityListWrapper.cities[0].name", equalTo("San Bernardino"))
+                .body("cityListWrapper.cities[0].country", equalTo("EEUU"));
+
+    }
+
+    @Test
+    public void testXMLPayloadPostRequest() throws JAXBException {
+
+        City city = new City("Paris", "France");
+        List<City> cityList = new ArrayList<>();
+        cityList.add(city);
+
+        CityListDTO requestCityList = new CityListDTO(cityList);
+
+        String xmlPayload = convertCityListDTOToXml(requestCityList);
+
+        String payload = app.given().contentType(MediaType.APPLICATION_XML)
+                .body(xmlPayload) // Use the generated XML payload
+                .when()
+                .post("/city")
+                .then()
+                .extract().asString();
+        Document doc = Jsoup.parse(payload);
+        String unescapedXml = doc.text();
+        assertThat(unescapedXml, equalTo(EXPECTED_XML_PAYLOAD));
+    }
+
+    @Test
+    public void testExpectedYamlResponse() {
+
+        final Response response = app.given()
+                .contentType("application/yaml")
+                .get("/city/getYamlFile")
+                .then()
+                .statusCode(HttpStatus.SC_OK)
+                .extract().response();
+        if (OS.current() == OS.WINDOWS) {
+            assertThat(response.asPrettyString(), is(EXPECTED_RESPONSE_BODY_YAML_WINDOWS_OS));
+        } else {
+            assertThat(response.asPrettyString(), is(EXPECTED_RESPONSE_BODY_YAML));
+        }
+
+    }
+
+    @Test
+    public void testPostYamlPayloadRequest() throws IOException {
+
+        String yamlString = readYamlFile("src/test/resources/payload.yaml");
+
+        Response response = RestAssured
+                .given()
+                .config(RestAssured.config()
+                        .encoderConfig(encoderConfig().encodeContentTypeAs("application/x-yaml", ContentType.TEXT)))
+                .contentType("application/x-yaml")
+                .body(yamlString)
+                .when()
+                .post("/city/cityYaml")
+                .then()
+                .statusCode(200).extract().response();
+        JsonPath jsonPath = response.jsonPath();
+
+        List<Object> cities = jsonPath.getList("expectedResponse.body.cities");
+        assertEquals("[{name=Tokio, country=Japan}]", cities.get(0).toString());
+    }
+
+    private String readYamlFile(String filePath) throws IOException {
+        Path path = Paths.get(filePath);
+        return new String(Files.readAllBytes(path));
+    }
+
+    private String convertCityListDTOToXml(CityListDTO cityListDTO) throws JAXBException {
```

Comment:
why not leave it to RestAssured? here https://www.james-willett.com/rest-assured-serialization-with-json-and-xml/#xml-serialization it looks like XML serialization should work.

Reply:
good example, it works.